server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalTopHitsTests.java

```diff
@@ -112,7 +112,8 @@ protected InternalTopHits createTestInstance(String name, List<PipelineAggregato
         } else {
             topDocs = new TopDocs(new TotalHits(totalHits, TotalHits.Relation.EQUAL_TO), scoreDocs);
         }
-        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore);
+        // Lucene's TopDocs initializes the maxScore to Float.NaN, if there is no maxScore
+        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore == Float.MIN_VALUE ? Float.NaN : maxScore);
```

Comment:
for correctness, maybe we should use another value than MIN_VALUE initially since it is a valid score value? eg. -1 or NEGATIVE_INFINITY, which are illegal since they are negative?